src/BlockTypes/MiniCart.php

```diff
@@ -322,11 +322,19 @@ protected function append_script_and_deps_src( $script ) {
 
 		$site_url = site_url() ?? wp_guess_url();
 
+		if ( version_compare( get_bloginfo( 'version' ), '6.2.2', '>' ) ) {
```

Comment:
This could cause an error if a version such as 6.2.3 is released before 6.3. I coincidentally opened a PR to solve this not being aware of this one: https://github.com/woocommerce/woocommerce-blocks/pull/10190. That PR was closed out in favor of this one, but notice how its a slightly different implementation that guards us against this possibility.

Reply:
Right. I see what you mean. Do you think I should update my PR?

I was trying searching through our repo (and WooCommerce) to find similar version checks, to follow the pattern. Seems usually it's a simple check that doesn't involve a regular expression.

Reply:
Would something like this work `version_compare( get_bloginfo( 'version' ), '6.3.0', '>=' )`? I think that would prevent any issues in case there is a 6.2.3 release and will also keep the code simple.

Reply:
I have found some issues checking against `6.3.0` when running beta versions of WordPress like `6.3-beta4-56226`.

Reply:
> Right. I see what you mean. Do you think I should update my PR?

Sure!

> I have found some issues checking against 6.3.0 when running beta versions of WordPress like 6.3-beta4-56226.

Right: that's the reason I've implemented like this on https://github.com/woocommerce/woocommerce-blocks/pull/10190 , it accounts for the beta version as well:

```php
$current_wp_version = get_bloginfo( 'version' );
if ( preg_match( '/^([0-9]+\.[0-9]+)/', $current_wp_version, $matches ) ) {
	$current_wp_version = (float) $matches[1];
}

if ( version_compare( $current_wp_version, '6.3', 'ge' ) ) {
	$before = $wp_scripts->get_inline_script_data( $script->handle, 'before' );
	$after  = $wp_scripts->get_inline_script_data( $script->handle );
} else {
	$before = $wp_scripts->print_inline_script( $script->handle, 'before', false );
	$after  = $wp_scripts->print_inline_script( $script->handle, 'after', false );
}
```

In general, we shouldn't need the ugly regex, but the `-beta` on specific versions made it a requirement 😬 : open to other ideas/suggestions if folks have any.

Reply:
I just noticed we have two ongoing discussions on the same topic 😆 . @roykho [mentioned below](https://github.com/woocommerce/woocommerce-blocks/pull/10165#discussion_r1262574561) we may not need to cover the `-beta` use-cases: if we don't support this, I suspect the only downside of this is that it will be difficult for us to test the changes made on this PR before the 6.3 official release (as that's the moment when `-beta` will be dropped).

Reply:
> So the reason why this is happening is because version_compare uses the semantic versioning. And WP is unorthodox and doesn't follow "semver". Because if they did, the version should look like 6.3.0-beta4-63233 and version_compare( $version, '6.3', '<' ) would cover that. But it won't cover unconventional 6.3-beta...

Yep, exactly!

Reply:
@danieldudzic if there's anything I can help you with the implementation here, just say the word! We will need the regex as demonstrated on https://github.com/woocommerce/woocommerce-blocks/pull/10165#discussion_r1262634275 otherwise it will be impossible to test this PR before the release of WordPress version 6.3 (as that's the only moment when the `-beta` will be dropped from the version number).

Reply:
Thank you for the input @nefeline and @roykho. I will make the update :)

Reply:
@danieldudzic if you're gonna go ahead with the regex path, then I would suggest you make a shared utilities function that can be used throughout the project as I am sure this will happen again and we don't want to have this version compare regex littered all over the place.

Reply:
Great idea. Let me add that now.